iceoryx_hoofs/source/posix_wrapper/shared_memory_object/memory_map.cpp

```diff
@@ -47,7 +47,7 @@ cxx::expected<MemoryMap, MemoryMapError> MemoryMapBuilder::create() noexcept
                                   m_fileDescriptor,
                                   m_offset)
                       // NOLINTNEXTLINE(cppcoreguidelines-pro-type-cstyle-cast, performance-no-int-to-ptr)
-                      .failureReturnValue(reinterpret_cast<void*>(MAP_FAILED))
+                      .failureReturnValue(MAP_FAILED)
```

Comment:
yes it is still needed since in the macro `#define MAP_FAILED (void*)-1` the c-style cast is hidden and this macro is provided by the system libraries.